src/Microsoft.Crank.Agent/Controllers/JobsController.cs

```diff
@@ -654,16 +654,29 @@ public async Task<IActionResult> Download(int id, string path)
                     return NotFound();
                 }
 
+                // Downloads can't get out of this path
+                var rootPath = Directory.GetParent(job.BasePath).FullName;
+
+                // Assume ~/ means relative to the project, otherwise it's relative to the application folder (job.BasePath)
+                var isRelativeToProject = String.IsNullOrEmpty(job.Source.DockerFile) && path.StartsWith("~/") || path.StartsWith("~\\");
```

Comment:
Why the check for if the DockerFile is empty? Does that mean it doesn't work with docker files?

Reply:
Not sure what it would mean for docker file, so I want to ignore it for now. Foo csprojs I know that a sub folder named 'published' is created, and it's the default location where we run apps. But for Docker it depends on each image what the current path is.

Reply:
Should we throw in that case then? To tell the user that this isn't possible?